### ISSUE TITLE
 [clang-tidy] `cppcoreguidelines-c-copy-assignment-signature` false positive with explicit object parameter

### ISSUE THREAD

**Issue by oracle-9** [llvm/llvm-project#107119]
The following code triggers `cppcoreguidelines-c-copy-assignment-signature` (aka `misc-unconventional-assign-operator`):
```cpp
struct Foo {
    Foo& operator=(this Foo&, Foo const&);
};
```
The check probably assumes the first parameter in the copy/move assignment operator is always the rhs. With the C++23 explicit object parameter feature, this is no longer always the case.

> `clang-tidy --version`
> Ubuntu LLVM version 18.1.3
>   Optimized build.
